packages/ra-supabase-ui-materialui/src/SocialAuthButton.tsx

```diff
@@ -0,0 +1,346 @@
+import * as React from 'react';
+import { Provider } from '@supabase/supabase-js';
+import { Button, ButtonProps } from '@mui/material';
+import { useLogin, useNotify, useTranslate } from 'ra-core';
+import {
+    AppleIcon,
+    AzureIcon,
+    BitbucketIcon,
+    DiscordIcon,
+    FacebookIcon,
+    GithubIcon,
+    GitlabIcon,
+    GoogleIcon,
+    KeycloakIcon,
+    LinkedinIcon,
+    NotionIcon,
+    SlackIcon,
+    SpotifyIcon,
+    TwitchIcon,
+    TwitterIcon,
+    WorkosIcon,
+} from './icons';
+
+export const SocialAuthButton = ({
+    provider,
+    redirect: redirectTo,
+    ...props
+}: SocialAuthButtonProps) => {
+    const login = useLogin();
+    const notify = useNotify();
+
+    const handleClick = () => {
+        login({ provider }, redirectTo ?? window.location.toString()).catch(
+            error => {
+                // The authProvide always reject for OAuth login but there will be no error
+                // if the call actually succeeds. This is to avoid react-admin redirecting
+                // immediately to the provided redirect prop before users are redirected to
+                // the OAuth provider.
+                if (error) {
+                    notify((error as Error).message, { type: 'error' });
+                }
+            }
+        );
+    };
+
+    return (
+        <Button
+            onClick={handleClick}
+            variant="contained"
+            size="medium"
+            color="inherit"
+            {...props}
+        />
+    );
+};
+
+export type SocialAuthButtonProps = {
+    provider: Provider;
+    redirect?: string;
+} & ButtonProps;
+
+export const AppleButton = (props: Omit<SocialAuthButtonProps, 'provider'>) => {
+    const translate = useTranslate();
+    const label = translate('ra-supabase.auth.sign_in_with', {
+        provider: 'Apple',
+    });
```

Comment:
Couldn't this part be factorized into `SocialAuthButton` too?

Reply:
It could but I'd rather avoid another abstraction